src/compiler/interpreter.rs

```diff
@@ -585,15 +586,11 @@ fn resolve_definition<'a>(
 }
 
 fn is_variable_name(name: &str) -> bool {
```

Comment:
But the function look the same just checking whether each char is valid
```
is_variable_name(      )...all(|c| c.is_ascii_uppercase()||c.is_ascii_digit()||c=='_');
is_valid_variable_name()...all(|c|{c.is_ascii_uppercase()||c.is_ascii_digit()||c=='_'
                               || !c.is_ascii()});}
```

So currently the following works regardless of whether you assign a number to a rule or a variable (it fails on identical assignment, but due to a different `kind` check)

```
main: ~(1 | 2);
2='var'
1: 'rule'{rule};
```

The new symbols, lacking upper/lower case properties, behave the same way like the numbers currently do (and fail on the same `kind` check in case of identical names for a rule and a var)